### ISSUE TITLE
Accessibility: Narrator cannot focus entire texts and announce "Document selected" when pressing Ctrl+A to choose all texts in ComboBox

### ISSUE THREAD

**Issue by Philip-Wang01** [dotnet/winforms#10007]
### .NET version

.Net 8.0: 8.0.100-rtm.23477.34

### Did it work in .NET Framework?

Yes

### Did it work in any of the earlier releases of .NET Core or .NET 5+?

No, this issue can repro in .Net 6.0 and .Net 7.0.

### Issue description

When using keyboard: Ctrl+A or mouse to choose all texts in ComboBox, Narrator cannot focus entire texts and announce "Document selected" to user.

The original issue: https://github.com/dotnet/winforms/issues/9828

The result in .Net 8.0 RC2 latest build:
![image](https://github.com/dotnet/winforms/assets/86937911/cbf91203-fa2d-428a-a16b-64b1ad2f5b6c)
![test80trm](https://github.com/dotnet/winforms/assets/86937911/b193cd15-df1d-4536-a89c-3120b73458af)

The result in .Net framework:
![image](https://github.com/dotnet/winforms/assets/86937911/9570f1a0-484e-45c4-91e2-bb0632531251)
![test481framework](https://github.com/dotnet/winforms/assets/86937911/6d23710f-ef74-4729-8964-3d97f71a2ec7)

### Steps to reproduce

1. Create a winforms .Net Core application.
2. Add a ComboBox control to form designer with text.
3. Build and run.
4. Use Narrator to focus on the individual text, then presse Ctrl+A to select all texts.

**Comment by Philip-Wang01** [comment]
@Tanya-Solyanik Verified this issue in main branch of Winforms repo, Narrator announce "**ComboBox text**" but not announce "**Document selected**" when pressing Ctrl+A to choose all texts in ComboBox, This is inconsistent with .net framework and TextBox/DomainUpDown/NumericUpDown, is this the expected result?
![image](https://github.com/dotnet/winforms/assets/86937911/42d3f93a-0d29-452b-9d3e-0552b44afc2d)

![test10007](https://github.com/dotnet/winforms/assets/86937911/8e29e176-3356-4237-af16-ac75998cc016)

**Comment by Philip-Wang01** [comment]
@Tanya-Solyanik Just a warm reminder, could you please have a look at your convenience?

**Comment by ricardobossan** [comment]
@Philip-Wang01 I just created WinForms apps for NetFramework and .NET Core and tried to reproduce the issue. It seems the behavior is the same in both:

### NetFramework App

![Framework_NotDocumentSelected](https://github.com/dotnet/winforms/assets/30190295/8f101910-ae04-4707-a25e-e68765194f28)

### .NET Core App

![Core_NotDocumentSelected](https://github.com/dotnet/winforms/assets/30190295/aea7bfaa-7978-4f28-8898-2d8f5057ab93)

**Comment by Philip-Wang01** [comment]
@ricardobossan Please select any letter first, for example "O". While making sure the narrator reads O, use the shortcut Ctrl+A to select all text. Then observe the result.
![image](https://github.com/dotnet/winforms/assets/86937911/450957d0-3e52-4794-8afe-4c4a3426c24f)
Don't use the mouse to click on the dropdown box, or switch other items in the comboBox. This issue is not related to the other items of the comboBox.
By the way, the [PR](https://github.com/dotnet/winforms/pull/10128) for this issue is in .NET 9.0, please set the `Target framework` to .NET 9.0 for validation.

**Comment by ricardobossan** [comment]
> @ricardobossan Please select any letter first, for example "O". While making sure the narrator reads O, use the shortcut Ctrl+A to select all text. Then observe the result. ![image](https://user-images.githubusercontent.com/86937911/280667660-450957d0-3e52-4794-8afe-4c4a3426c24f.png) Don't use the mouse to click on the dropdown box, or switch other items in the comboBox. This issue is not related to the other items of the comboBox. By the way, the [PR](https://github.com/dotnet/winforms/pull/10128) for this issue is in .NET 9.0, please set the `Target framework` to .NET 9.0 for validation.

I see now the inconsistency that you mentioned. When pressing `ctrl+a` the behavior does differ from .net framework. I'll ask @Tanya-Solyanik about this.

**Comment by Tanya-Solyanik** [comment]
@ricardobossan @Philip-Wang01  - do you see differences in NVDA and JAWS as well between .NET Framework and .NET9. Also, what happens in WPF in the same scenario?

Personally, I like that the new behavior does not depend on whether the previous selection existed (I think it's confusing when the same content selected in 2 different ways, it's announced differently, i.e. after selection `o` we head "document selected"  but when selecting the first time, we head the actual content.)

**Comment by Philip-Wang01** [comment]
@Tanya-Solyanik @ricardobossan 
**1:** .NET Framework and .NET 9.0 have the same results in NVDA, NVDA announce "ComboBox text" but not announce "Document selected" when pressing Ctrl+A to choose all texts in ComboBox.
![image](https://github.com/dotnet/winforms/assets/86937911/509ad057-cfc0-4080-b8f5-41080782515c)

**2:** .NET Framework and .NET 9.0 have the same results in JAWS, JAWS will not announce any results when pressing Ctrl+A to choose all texts in ComboBox.

**3:** In WPF project, Narrator announce "Document selected" but not announce "ComboBox text" when pressing Ctrl+A to choose all texts in ComboBox.
![image](https://github.com/dotnet/winforms/assets/86937911/337a65ec-748b-4dff-83b0-a7a9d50232b9)

**Comment by ricardobossan** [comment]
@Tanya-Solyanik @Philip-Wang01 

Hi, @Merrie. How do you do?

As discussed in the comments above[¹](https://github.com/dotnet/winforms/issues/10007#issuecomment-1778436509) [²](https://github.com/dotnet/winforms/issues/10007#issuecomment-1797049336) [³](https://github.com/dotnet/winforms/issues/10007#issuecomment-1797860808), It seems my fix to this issue has introduced behavior that differs from the one presented by .Net Framework when `ctrl+a` is pressed on a ComboBox.

My fix, [PR 10128](https://github.com/dotnet/winforms/pull/10128) makes narrator say the name of the selected item in the ComboBox:

![image](https://github.com/dotnet/winforms/assets/30190295/9835dea1-28c1-4042-9c0a-e71bf8184e3f)

However, in NetFramework, narrator would just say Document selected, like so:

![image](https://github.com/dotnet/winforms/assets/30190295/8f6ecca0-93c5-4c04-aa9d-6dc48cbd6682)


Which would be the desired outcome? Should I reopen the issue and make it so that narrator says Document selected when ctrl+a is pressed on the ComboBox?

**Comment by merriemcgaw** [comment]
I don't think making it say "document selected" is really that helpful in a ComboBox. I don't believe we need another issue myself.
